test/e2e/apibinding/apibinding_permissionclaims_test.go

```diff
@@ -0,0 +1,229 @@
+
+	kcpClients, err := clientset.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+
+	dynamicClients, err := dynamic.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct dynamic cluster client for server")
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct dynamic cluster client for server")
+
+	apifixtures.CreateSheriffsSchemaAndExport(ctx, t, serviceProviderWorkspace, kcpClients, "wild.wild.west", "board the wanderer")
+
+	identityHash := ""
+	//TODO: convert to framework.Eventually
+	framework.Eventually(t, func() (done bool, str string) {
+		sheriffExport, err := kcpClients.Cluster(serviceProviderWorkspace).ApisV1alpha1().APIExports().Get(ctx, "wild.wild.west", metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+
+		if conditions.IsTrue(sheriffExport, apisv1alpha1.APIExportIdentityValid) {
+			identityHash = sheriffExport.Status.IdentityHash
+			return true, "found"
+		}
+		return false, "not found"
+
+	}, wait.ForeverTestTimeout, 100*time.Millisecond)
+
+	t.Logf("Found identity hash: %v", identityHash)
+	apifixtures.BindToExport(ctx, t, serviceProviderWorkspace, "wild.wild.west", consumerWorkspace, kcpClients)
+
+	t.Logf("set up service provider with permission claims")
+	setUpServiceProviderWithPermissionClaims(ctx, dynamicClients, kcpClients, kubeClusterClient, serviceProviderWorkspace, t, identityHash)
+
+	t.Logf("set up binding, with invalid accepted claims hash")
+	bindConsumerToProvider(ctx, consumerWorkspace, serviceProviderWorkspace, t, kcpClients, "xxxxxxx")
+
+	//validate the the indenitty mismatch condition occurs
+	t.Logf("validate that the permission claim's conditions are false and invalid identity hash is the reason")
+
+	framework.Eventually(t, func() (bool, string) {
+		// get the binding
+		binding, err := kcpClients.Cluster(consumerWorkspace).ApisV1alpha1().APIBindings().Get(ctx, "cowboys", metav1.GetOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+
+		cond := conditions.Get(binding, apisv1alpha1.AcceptedPermissionClaimValid)
+		if cond == nil {
+			return false, ""
+		}
+
+		if cond.Status == v1.ConditionFalse && cond.Reason == apisv1alpha1.IdentityMismatchClaimInvalidReason {
+			return true, ""
+		}
+		return false, ""
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "unable to see invalid identity hash")
+
+	t.Logf("update to correct hash")
+	binding, err := kcpClients.Cluster(consumerWorkspace).ApisV1alpha1().APIBindings().Get(ctx, "cowboys", metav1.GetOptions{})
+	require.NoError(t, err)
+	binding.Spec.AcceptedPermissionClaims = getAcceptedPermissionClaims(identityHash)
+	_, err = kcpClients.Cluster(consumerWorkspace).ApisV1alpha1().APIBindings().Update(ctx, binding, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Validate that the permission claims are valid")
+	framework.Eventually(t, func() (bool, string) {
```

Comment:
be verbose about failures

Reply:
updated all the other framework.Eventuallys